### PR TITLE
Add timeout to all API requests and more logging

### PR DESCRIPTION
**Pull request by austinkelleher**

### CODE REVIEW

File: src/steps/constants.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ export const Relationships: Record<
     targetType: Entities.MAC_OS_CONFIGURATION_PROFILE._type,
   },
   COMPUTER_USES_PROFILE: {
-    _type: 'user_endpoint_uses_osx_configuration_profile',
+    _type: 'user_endpoint_uses_jamf_osx_configuration_profile',
```
Comment (austinkelleher) [inline src/steps/constants.ts:126]: This was just incorrect.

File: src/steps/users/index.ts [hunk 2]
```diff
@@ -13,13 +15,16 @@ import { createAdminEntity, createDeviceUserEntity } from './converters';
...
+import { wrapWithTimer } from '../../util/timer';
 
 async function iterateAdminUserProfiles(
   client: JamfClient,
+  logger: IntegrationLogger,
   jobState: JobState,
   iteratee: (user: Admin) => Promise<void>,
 ) {
   const accountData = await getAccountData(jobState);
+  logger.info({ numAdmins: accountData.users }, 'Iterating account admins');
```
Comment (aiwilliams) [inline src/steps/users/index.ts:27]: Do u mean for this to be .length ?

File: src/util/timer.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { IntegrationLogger } from '@jupiterone/integration-sdk-core';
+
+export interface WrapWithTimerParams {
+  operationName: string;
+  logger: IntegrationLogger;
+  metadata?: Record<string, any>;
+}
+
+export async function wrapWithTimer<T>(
```
Comment (aiwilliams) [inline src/util/timer.ts:9]: I pine for logOperation as a name.

**Review by aiwilliams — DISMISSED** [review]

File: src/steps/users/index.ts (hunk 2 above)
Comment (austinkelleher) [reply src/steps/users/index.ts:27]: Thanks for catching this!

File: src/util/timer.ts (hunk 3 above)
Comment (austinkelleher) [reply src/util/timer.ts:9]: `wrapWithTimer` is the same function name that we use internally, which is why I chose it. I don't feel strongly about this though.

Comment (austinkelleher) [reply src/util/timer.ts:9]: Will do this in a separate PR if we decide to make the change.

Comment (aiwilliams) [reply src/util/timer.ts:9]: No need, but thanks for considering! Wouldn't want to cause confusion.

**Review by aiwilliams — APPROVED** [review]

**Pull request merged**